src/Shared/ServerInfrastructure/StringUtilities.cs

```diff
@@ -387,28 +402,54 @@ public static string ConcatAsHexSuffix(string str, char separator, uint number)
                 length += str.Length;
             }
 
-            return string.Create(length, (str, separator, number), (buffer, tuple) =>
+            return string.Create(length, (str, separator, number), s_populateSpanWithHexSuffix);
+        }
+
+        private static void PopulateSpanWithHexSuffix(Span<char> buffer, (string str, char separator, uint number) tuple)
+        {
+            var (tupleStr, tupleSeparator, tupleNumber) = tuple;
+
+            var i = 0;
+            if (tupleStr != null)
             {
-                var (tupleStr, tupleSeparator, tupleNumber) = tuple;
-                char[] encode16Chars = s_encode16Chars;
+                tupleStr.AsSpan().CopyTo(buffer);
+                i = tupleStr.Length;
+            }
 
-                var i = 0;
-                if (tupleStr != null)
-                {
-                    tupleStr.AsSpan().CopyTo(buffer);
-                    i = tupleStr.Length;
-                }
+            buffer[i] = tupleSeparator;
+            i++;
 
-                buffer[i + 8] = encode16Chars[tupleNumber & 0xF];
-                buffer[i + 7] = encode16Chars[(tupleNumber >> 4) & 0xF];
-                buffer[i + 6] = encode16Chars[(tupleNumber >> 8) & 0xF];
-                buffer[i + 5] = encode16Chars[(tupleNumber >> 12) & 0xF];
-                buffer[i + 4] = encode16Chars[(tupleNumber >> 16) & 0xF];
-                buffer[i + 3] = encode16Chars[(tupleNumber >> 20) & 0xF];
-                buffer[i + 2] = encode16Chars[(tupleNumber >> 24) & 0xF];
-                buffer[i + 1] = encode16Chars[(tupleNumber >> 28) & 0xF];
-                buffer[i] = tupleSeparator;
-            });
+            if (Ssse3.IsSupported)
+            {
+                var lowNibbles = Ssse3.Shuffle(Vector128.CreateScalar(tupleNumber).AsByte(),s_shuffleMask);
```

Comment:
```suggestion
                var lowNibbles = Ssse3.Shuffle(Vector128.CreateScalarUnsafe(tupleNumber).AsByte(), s_shuffleMask);
```
Should work too, as on L426 garbage is masked out anyway.